pytensor/tensor/basic.py

```diff
@@ -760,7 +760,12 @@ def switch(cond, ift, iff):
     """if cond then ift else iff"""
 
 
-where = switch
+# where = switch
+def where(cond, ift, iff):
+    if ift is not None and iff is not None:
+        return switch(cond, ift, iff)
+    else:
+        return cond.nonzero()
```

Comment:
```suggestion
        return as_tensor(cond).nonzero()
```

Reply:
btw we arent supposed to use `.eval()` within this function definition right? i should only do that to actually check for the values in the test.
also, in case where only condition is passed, the output is actually a tuple but we only actually require the first input so should i index it with [0] or no. the numpy output is also a tuple but in our case, we can only run `.eval()` on the first value